src/java/fr/paris/lutece/portal/service/security/SecurityService.java

```diff
@@ -311,6 +311,12 @@ private static LuteceAuthentication getPortalAuthentication( ) throws LuteceInit
     public void registerUser( HttpServletRequest request, LuteceUser user )
     {
         HttpSession session = request.getSession( true );
+
+        // invalidate the current session
+        session.invalidate( );

```

Comment:
Wouldn't it be safer to only invalidate if there was a previous `ATTRIBUTE_LUTECE_USER` registered ? If not, the user might loose some info gathered while not connected.

Reply:
You're right, we will loose session data, such as selected language.
But to avoid session fixation, we need to renew the session. If we keep the session for a "first authentication", we don't deal with the problem.
We could get the current selected language and set this attribute in the new session. 
Maybe we could manage in properties a list of attributes to save and restore in the new session, to allow specific configuration by override ?

Reply:
What is the risk of bringing over all attributes ? This would be equivalent to today's situation minus the session fixation issue.

Reply:
request.changeSessionId() should do the trick without modifying session data

Reply:
Cool, though that makes a Servlet 3.1 container a requirement (tomcat >= 8), which should be appropriatly communicated